### PR TITLE
Don't reset screen on focus gain events

### PR DESCRIPTION
**Pull request by johnnovak**
On Windows 10 when using DOSBox in windowed mode, there's some quite audible sound stuttering and display flashing when the DOSBox window regains focus. The sound stutter is basically the same stutter that happens when switching between windowed and fullscreen modes, so it's quite bad.

Now, this is a big problem for me personally because when I play oldschool grid-based RPGs, I always use windowed mode: on the left half of the screen is the DOSBox window, and on the right my map editor. Naturally, I keep alt-tabbing between the two quite vigorously as I'm mapping out a dungeon. Other emulators that I use the same way, such as WinUAE, don't have this problem.

@kcgen I saw that it was you who put this in; I've read your comment a few times and have concluded that this is not needed as `GFX_ResetScreen` is always called in `GFX_SwitchFullScreen` anyway. So why we should call it twice, I don't understand. Maybe it's required on Linux for some obscure reason, I really can't tell. All I know is that by removing this forced gfx reset on window focus & exposed events, my sound stutter and display flash issue is 100% gone. So at the very least I'd like to ifdef it out for the Windows builds.

Happy to have a further look at edge cases on my box if you tell me what to test specifically, but I couldn't notice any problems with the few games I tried.

### CODE REVIEW

File: src/gui/sdlmain.cpp [hunk 1]
```diff
@@ -3691,19 +3691,6 @@ bool GFX_Events()
...
-				// When we're fullscreen, the DOS program might
-				// change underlying draw resolutions, which can
-				// pose a problem when switching back to
-				// window-mode (beacuse the prior window-size
-				// may not longer accomodate the new draw-size).
-				// So in this case, we want to reset the screen
-				// size - but only if we're not in the middle of
-				// resizing the window (which also fires EXPOSED
-				// events).
-				if (!sdl.desktop.fullscreen &&
```
Comment (johnnovak) [inline src/gui/sdlmain.cpp:3703]: What's strange about this comment is this part: "When we're fullscreen, the DOS program might change underlying draw resolutions". Well, this `GFX_ResetScreen` will only happen if we're _not_ in fullscreen.

Comment (kcgen) [reply src/gui/sdlmain.cpp:3703]: The scenario involved starting fullscreen in a graphical mode (say, 320x200), then while still fullscreen, exiting back out to a text mode (larger resolution), and then toggling out of fullscreen.

@bmunger reported it, exiting out of (either Quake or Blood?) - where the text screen was cutoff a bit.
It was easily reprodiceable at the time, but now I can reproduce it (and tried many resolutions and output modes).  

If it can go away, then all the better! The fewer corner cases in SDL main the better!

@bmunger - maybe you can remember the scenario better?

**Comment by kcgen** [comment]
I can't see any issues on my side, and have covered all the output types and size changes (upward and downward in resolution). Likewise, tests are passing on your side under Windows. 
Merging; we can always bisect back to this if there's a strange corner case that this was still handling.
